app/src/main/java/com/teamagam/gimelgimel/app/message/view/SendImageFragment.java

```diff
@@ -49,23 +57,50 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     }
 
     @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putString(CAPTURED_IMAGE_PATH_KEY, mCapturedImagePath);
+    }
+
+    @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == SendImageMessageViewModel.REQUEST_IMAGE_CAPTURE) {
+        if (requestCode == REQUEST_IMAGE_CAPTURE) {
             if (isImageCaptured(resultCode)) {
                 sLogger.userInteraction("Sending camera activity result");
 
-                mViewModel.sendImage();
+                mViewModel.sendImage(mCapturedImagePath);
             } else {
                 sLogger.userInteraction("Camera activity returned with no captured image");
             }
         }
     }
 
+    public void takePicture() {
+        Uri newImageUri = createNewImageUri();
+        mCapturedImagePath = newImageUri.getPath();
+        startCameraIntent(newImageUri);
+    }
+
     @Override
     protected int getFragmentLayout() {
         return R.layout.fragment_send_image;
     }
 
+
+    private Uri createNewImageUri() {
+        return ImageUtils.getTempImageUri(getContext());
```

Comment:
ImageUtils is in data module. Are you sure this is a good idea to call it directly?